src/modm/driver/inertial/adis16470.hpp

```diff
@@ -0,0 +1,233 @@
+			return AccessMethod::Read;
+		}
+		if ((r >= Register::FIRM_REV) && (r != Register::USER_SCR1) && (r != Register::USER_SCR2) && (r != Register::USER_SCR3)) {
+			return AccessMethod::Read;
+		}
+		return AccessMethod::ReadWrite;
+	}
+
+	/// Status/Error Flag Indicators
+	/// DIAG_STAT register (address 0x02, 0x03) reset value is 0x0000
+	enum class
+	DiagStat : uint16_t
+	{
+		// Reserved				= Bit8 to Bit15,
+		ClockError				= Bit7, ///< Indicates that the internal data sampling clock does not synchronize with the external clock, which only applies when using scale sync mode.
+		MemoryFailure			= Bit6, ///< Indicates a failure in the flash memory test.
+		SensorFailure			= Bit5, ///< Indicates failure of at least one sensor, at the conclusion of the self test.
+		StandbyMode				= Bit4, ///< Indicates that the voltage across VDD and GND is <2.8 V, which causes data processing to stop.
+		SpiCommunicationError	= Bit3, ///< Indicates that the total number of SCLK cycles is not equal to an integer multiple of 16.
+		FlashUpdateFailure		= Bit2, ///< Indicates that the most recent flash memory update failed.
+		DataPathOverrun			= Bit1, ///< Indicates that one of the data paths have experienced an overrun condition.
+		// Reserved				= Bit0,
+	};
+	MODM_FLAGS8(DiagStat);
+
+};
+
+/**
+ * \ingroup	modm_driver_adis16470
+ * \author	Raphael Lehmann
+ */
+template<class SpiMaster, class Cs>
+class Adis16470 : public adis16470,
+				  public modm::SpiDevice<SpiMaster>,
+				  protected modm::NestedResumable<2>
+{
+public:
+	Adis16470() = default;
+
+	/**
+	 * @brief Initialize
+	 *
+	 * Sets used pins as output.
+	 *
+	 * SPI must be initialized by the user!
+	 * @warning The SPI frequency must not exceed 2 MHz for this chip,
+	 * or 1 MHz with burst mode.
+	 */
+	modm::ResumableResult<void>
+	initialize();
+
+	/**
+	 * @brief Read a single register.
+	 *
+	 * @param reg The register to be read
+	 * @return The register value in case of a read access, or std::nullopt if
+	 * an error occured, e.g. if some register access is not permitted.
+	 */
+	modm::ResumableResult<std::optional<uint16_t>>
+	readRegister(Register reg);
+
+	/**
+	 * @brief Write a single register.
+	 *
+	 * @param reg The register to be written
+	 * @param value The value to be written to the register.
+	 * @return False in case of any error, e.g. if some register acces is not
+	 * permitted.
+	 */
+	modm::ResumableResult<bool>
+	writeRegister(Register reg, uint16_t value);
+
+	/**
+	 * @brief Read a sequence of registers.
+	 * Improves SPI usage efficiency by 100% compared to `accessRegister(...)`.
+	 *
+	 * @param sequence All registers in the std::span will be read and the
+	 * results will be stored in values.
+	 * @param values Results of the read operations
+	 * @return False in case of any error, e.g. if some register access is not
+	 * permitted.
+	 */
+	modm::ResumableResult<bool>
+	readRegisterSequence(std::span<Register> sequence, std::span<uint16_t> values);
```

Comment:
```suggestion
	readRegisterSequence(std::span<const Register> sequence, std::span<uint16_t> values);
```
The register list should be const.